### ISSUE TITLE
Add animations when inserting or removing items from the lists

### ISSUE THREAD

**Issue by Tiebo** [hanshengchiu/reorderables#23]
I miss the feature that items not only animate when being reordered but also when entering and leaving the list.

Is this possible?

**Comment by JonathanSiddle** [comment]
Hey! Did you manage to get this working? Does anyone have any pointers about how to implement this?

**Comment by joielechong** [comment]
This library has the capability to animate when item is deleted:
https://github.com/fluttercandies/flutter_draggable_container

But the main problem is, afaik, there is no capability to remove item manually.

I'm stick to use reorderables library for now, though the animation is nice to have.

**Comment by RaviKavaiya** [comment]
@hanshengchiu any updates??

**Comment by Tiebo** [comment]
@RaviKavaiya hey, check out this package, it does exactely this.

https://pub.dev/packages/implicitly_animated_reorderable_list

**Comment by RaviKavaiya** [comment]
@Tiebo that is good. But, that package doesn't work with Slivers..

**Comment by HeebeLee** [comment]
I also like this feature. Now, flutter has a lot of animation list components, including SliverAnimatedList, AnimatedList. Can it be updated so that we can insert data and delete data in animation?

@hanshengchiu

**Comment by hanshengchiu** [comment]
> I also like this feature. Now, flutter has a lot of animation list components, including SliverAnimatedList, AnimatedList. Can it be updated so that we can insert data and delete data in animation?
> 
> @hanshengchiu

@HeebeLee do you know how to do it?

**Comment by HeebeLee** [comment]
> > I also like this feature. Now, flutter has a lot of animation list components, including SliverAnimatedList, AnimatedList. Can it be updated so that we can insert data and delete data in animation?
> > @hanshengchiu
> 
> @HeebeLee do you know how to do it?

No, I don't know what to do. Because it requires a lot of familiarity with the two components, which is a little too hard for me.

I think it should be easier for you. Maybe you can find ideas in the AnimatedList. If you can, please take some time to finish it. Thank you.

**Comment by hanshengchiu** [comment]
Maybe you could try to replace the base class with SliverAnimatedList. I'm not sure if it's gonna work.
